assets/js/handpicked-products.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+import {
+	BlockAlignmentToolbar,
+	BlockControls,
+	InspectorControls,
+} from '@wordpress/editor';
+import {
+	Button,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	Spinner,
+	Toolbar,
+	withSpokenMessages,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import getQuery from './utils/get-query';
+import { IconWidgets } from './components/icons';
+import ProductsControl from './components/products-control';
+import ProductOrderbyControl from './components/product-orderby-control';
+import ProductPreview from './components/product-preview';
+
+/**
+ * Component to handle edit mode of "Hand-picked Products".
+ */
+class ProductsBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			products: [],
+			loaded: false,
+		};
+
+		this.debouncedGetProducts = debounce( this.getProducts.bind( this ), 200 );
+	}
+
+	componentDidMount() {
+		this.getProducts();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const hasChange = [ 'products', 'columns', 'orderby' ].reduce(
+			( acc, key ) => {
+				return acc || prevProps.attributes[ key ] !== this.props.attributes[ key ];
+			},
+			false
+		);
+		if ( hasChange ) {
+			this.debouncedGetProducts();
+		}
+	}
+
+	getProducts() {
+		if ( ! this.props.attributes.products.length ) {
```

Comment:
I see `this.props.attributes` is required, but any chance the `.products` key might not be set here?